### PR TITLE
Close coverage gaps and validate `theta_fitted` on assignment

### PR DESCRIPTION
**Pull request by michaelosthege**
This PR increases overall test coverage and adds a validation of `theta` shape on assignment of `theta_fitted`.

Many tests had to be updated because they assigned a vector of incorrect length.

### CODE REVIEW

File: calibr8/tests.py [hunk 1]
```diff
@@ -516,7 +600,7 @@ def test_assert_version_match(self):
     @pytest.mark.parametrize("residual_type", ["relative", "absolute"])
     def test_plot_model(self, residual_type):
         em = _TestPolynomialModel(independent_key='S', dependent_key='A365', mu_degree=1, scale_degree=1)
-        em.theta_fitted = [0, 2, 0.1, 1]
+        em.theta_fitted = [0, 2, 0.1, 1, 2]
```
Comment (lhelleckes) [inline calibr8/tests.py:603]: Can you please explain why all theta have an entry more? Is this the bug you mentioned?

Comment (michaelosthege) [reply calibr8/tests.py:603]: Yes, the number I added is the degree of freedom. All these tests passed the incorrect number of parameters.

Comment (lhelleckes) [reply calibr8/tests.py:603]: :O

**Review by lhelleckes — APPROVED** [review]
Ok

**Pull request merged**